src/riak_core_coverage_fsm.erl

```diff
@@ -87,28 +85,30 @@
          terminate/3,
          code_change/4]).
 
--spec behaviour_info(atom()) -> 'undefined' | [{atom(), arity()}].
-behaviour_info(callbacks) ->
-    [
-     {init, 2},
-     {process_results, 2},
-     {finish, 2}
-    ];
-behaviour_info(_) ->
-    undefined.
+-type mod_state() :: term().
+-callback init(from(), RequestArgs :: [term()]) ->
+    {Request :: term(), vnode_selector(), NVal :: pos_integer(), primary_vnode_coverage(),
+     NodeCheckService :: module(), VNodeMaster :: atom(), timeout(), mod_state()}.
+-callback process_results(Results :: term(), mod_state()) ->
+    {ok, mod_state()} |
+    {done, mod_state()} |
+    {error, term()}.
```

Comment:
That would be shorter, but I don't feel like it's clearer: there are three distinct types of result that `process_results` can return, so having each one split out into its own top-level type in the spec feels more intuitive to me.

We can squash the first two together, but only because they happen to bear a superficial resemblance in their structure, not because they necessarily share anything else in common.